agents/rust/mediator/mediation/.github/workflows/rust-basic.yml

```diff
@@ -0,0 +1,130 @@
+name: Rust basic workflow
+
+on:
+  push:
+    branches: [ "main", "ci-integration-test", "ci-cd", "dev", aries-vcx-mediator-integration]
+  pull_request:
+    branches: [ "main", "ci-integration-test", "ci-cd", "dev", aries-vcx-mediator-integration]
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  check:
+    name: Check for obvious errors
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+    - name: Cargo cache
+      uses: actions/cache@v3
+      with:
+        path: |
+          ~/.cargo/bin/
+          ~/.cargo/registry/index/
+          ~/.cargo/registry/cache/
+          ~/.cargo/git/db/
+          target/
+        key: ${{ runner.os }}-cargo-check-cache-${{ hashFiles('**/Cargo.lock') }}
+        restore-keys: |
+          ${{ runner.os }}-cargo-
+    - name: cargo check
+      run: cargo check --verbose
+  build:
```

Comment:
formatting: we have convention to leave blank line between jobs, see our main GA file